internal/config/types.go

```diff
@@ -125,3 +128,16 @@ type InstanceWatcher struct {
 type InstanceHealthWatcher struct {
 	MonitoringFrequency time.Duration `yaml:"-" mapstructure:"monitoring_frequency"`
 }
+
+func (c *Config) IsDirectoryAllowed(directory string) bool {
+	isDirectoryAllowed := false
+
+	for _, allowedDirectory := range c.AllowedDirectories {
+		if strings.HasPrefix(directory, allowedDirectory) {
+			isDirectoryAllowed = true
+			break
+		}
+	}
+
+	return isDirectoryAllowed
+}
```

Comment:
style: the variable and `break` aren't necessary imo.
```suggestion
func (c *Config) IsDirectoryAllowed(directory string) bool {
	for _, allowedDirectory := range c.AllowedDirectories {
		if strings.HasPrefix(directory, allowedDirectory) {
			return true
		}
	}

	return false
}
```

Reply:
This implementation assumes that `directory` and `allowedDirectory` are both absolute paths. Will that always be the case?

Reply:
yes both have to be absolute paths. If they were relative paths I dont think we would have a way of finding out what directory they are in.